diff --git a/scripts/yodl2whatever.in b/scripts/yodl2whatever.in
index cd58ba0..2096b81 100755
--- a/scripts/yodl2whatever.in
+++ b/scripts/yodl2whatever.in
@@ -87,10 +87,10 @@ fi
 base=`basename $0`
 
 # Determine the options, using getopt(1). See its man-page
-getopt -T > /dev/null
+@getopt@ -T > /dev/null
 if [ "$?" == "4" ] ; then
     # Rich man's getopt.
-    TEMP=`getopt -o D:d:ghi:I:kl:Lm:n:o:p:r:tVvWw \
+    TEMP=`@getopt@ -o D:d:ghi:I:kl:Lm:n:o:p:r:tVvWw \
         --long define:,definemacro:,help,include:,index:,keep-ws,live-data:\
         --long messages:,max-nested-files:,output:,preload:,max-replacements:\
         --long trace,version,no-warnings,verbose,warranty,warn,intermediate:\
@@ -98,7 +98,7 @@ if [ "$?" == "4" ] ; then
         -- "$@"`
 else
     # Poor man's getopt. Only single-char flags supported.
-    TEMP=`getopt D:d:ghi:I:kl:m:n:o:p:r:tVvWw  -- "$@"`
+    TEMP=`@getopt@ D:d:ghi:I:kl:m:n:o:p:r:tVvWw  -- "$@"`
 fi
 
 if [ $? != 0 ] ; then echo "getopt error in $base" ; exit 1 ; fi
