diff --git a/spch/process/process.h b/spch/process/process.h
index a1e68de..a8f1c7a 100644
--- a/spch/process/process.h
+++ b/spch/process/process.h
@@ -2,6 +2,7 @@
 #define INCLUDED_PROCESS_
 
 #include <iosfwd>
+#include <chrono>
 
 #ifdef fbb
     #include <bobcat/datetime>
diff --git a/spch/process/softlinks.cc b/spch/process/softlinks.cc
index 5bf58c1..e40b8df 100644
--- a/spch/process/softlinks.cc
+++ b/spch/process/softlinks.cc
@@ -13,6 +13,6 @@ int Process::softLinks() const
             return 1;
     }
 
-    fs::last_write_time(spchGch, fs::__file_clock::now());
+    fs::last_write_time(spchGch, std::chrono::file_clock::now());
     return 0;
 }
